Extensions/Guardrail+Settings.swift

```diff
@@ -125,12 +125,23 @@ public extension Guardrail where Value == HKQuantity {
 
         } else {
             return Guardrail(
-                absoluteBounds: supportedBasalRates.first!...absoluteUpperBound,
-                recommendedBounds:  supportedBasalRates.first!...absoluteUpperBound,
+                absoluteBounds: supportedBasalRates.drop { $0 <= 0 }.first!...absoluteUpperBound,
+                recommendedBounds:  supportedBasalRates.drop { $0 <= 0 }.first!...absoluteUpperBound,
                 unit: .internationalUnitsPerHour
             )
         }
     }
+    
+    static func selectableBasalRates(supportedBasalRates: [Double],
+                                     scheduledBasalRange: ClosedRange<Double>?,
+                                     lowestCarbRatio: Double?,
+                                     maximumBasalRatePrecision decimalPlaces: Int = 3) -> [Double] {
+        let basalGuardrail = Guardrail.maximumBasalRate(supportedBasalRates: supportedBasalRates, scheduledBasalRange: scheduledBasalRange, lowestCarbRatio: lowestCarbRatio)
+        let maximumScheduledBasalRate = scheduledBasalRange?.upperBound ?? -Double.infinity
+        return supportedBasalRates
+            .drop { $0 < maximumScheduledBasalRate }
```

Comment:
[nit] I'm confused by this. I read this whole thing as "the selectable basal rates are the supported basal rates, but NOT those that are less than the maximum scheduled basal rate AND those outside of the bounds of the basal guardrails. Why are we dropping those that are **less than** the maximum scheduled basal rate?

Reply:
This logic was copied from `DeliveryLimitsEditor.swift`, [old line 47](https://github.com/tidepool-org/LoopKit/pull/280/files/289ba59abf0325759b56ddf159f82090c3b3a197#diff-8c86c8f43c935b047878c22fd54650f5c064bea4c982a8fa6d1d18c51b5aebe2L47).  

This is honoring the following guardrail **Done** Criteria (from [LOOP-1732](https://tidepool.atlassian.net/browse/LOOP-1732)):

> **Max Basal Delivery Limit**
> ...
> Minimum:   Highest Scheduled Basal Rate
>

Reply:
Yes, your above comment makes sense. The *minimum* Max Basal Delivery Limit should be whatever the Highest Scheduled Basal Rate is currently. To allow the Max Basal to go below this Highest Schedule would cause a conflict.

However, this function is called `selectableBasalRates`, which is where my confusion stems from. If this was called `selectableMaxBasalRates` (note the addition of "Max") then it would make more sense to me.

Or am I completely misunderstanding something here?

Reply:
Oh oh, yeah that's a good point.  TBH I didn't think much about that, I just copied the name right out of `DeliveryLimitsEditor.swift`.  Moving it here means it deserves a new name, you are right 👍